### PR TITLE
No method reference on new class select

### PR DESCRIPTION
**Pull request by timtebeek**
As shared instance might behave different from new instances

### CODE REVIEW

File: src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java [hunk 1]
```diff
@@ -1290,4 +1290,32 @@ private Integer foo(String bar) {
...
+                  }
+              }
+              """
+          ),
+          java(
+            """
+              import java.util.stream.Stream;
+              class B {
+                  void bar(Stream<String> stream) {
+                      stream.map(s -> new A().lower(s));
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java:1314]: This used to insert the following, but that would then share the instance across operations, inflating the `seen` counter above.
```
                      stream.map(new A()::lower);
```

Comment (knutwannheden) [reply src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java:1314]: @timtebeek I believe the same problem applies if the select is a method invocation. Imagine having `s -> a().lower(s)` with
```java
    A a() {
        return new A();
    }
```

Comment (knutwannheden) [reply src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java:1314]: Turned out that case was already covered.

Comment (timtebeek) [reply src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java:1314]: Thanks for the rework! Hadn't spotted that other one, but like how you've solved it now.

**Pull request merged**